cmd/gen.go

```diff
@@ -1,80 +1,61 @@
 package cmd
 
 import (
-	"strings"
+	"io/ioutil"
 
 	"github.com/MakeNowJust/heredoc"
-	"github.com/odpf/meteor/generator"
+	"github.com/odpf/meteor/recipe"
 	"github.com/odpf/salt/log"
 	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
 )
 
 // GenCmd creates a command object for the "gen" action
 func GenCmd(lg log.Logger) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "gen",
-		Short: "Generate sample recipes and plugins",
-		Annotations: map[string]string{
-			"group:core": "true",
-		},
-	}
-
-	cmd.AddCommand(GenRecipeCmd())
-
-	return cmd
-}
-
-// GenRecipeCmd creates a command object for generating recipes
-func GenRecipeCmd() *cobra.Command {
 	var (
-		extractor  string
-		sinks      string
-		processors string
+		outputDirPath string
+		dataFilePath  string
 	)
 
 	cmd := &cobra.Command{
-		Use:     "recipe [name]",
-		Aliases: []string{"r"},
-		Args:    cobra.ExactValidArgs(1),
-		Short:   "Generate a new recipe",
+		Use:   "gen",
+		Args:  cobra.ExactValidArgs(1),
+		Short: "Generate recipes",
 		Long: heredoc.Doc(`
-			Generate a new recipe.
+			Generate multiple recipes using a template and list of data.
 
-			The recipe will be printed on standard output.
-			Specify recipe name with the first argument without extension.
-			Use commma to separate multiple sinks and processors.`),
+			The generated recipes will be created on output directory.`,
+		),
 		Example: heredoc.Doc(`
-			# generate a recipe with a bigquery extractor and a console sink
-			$ meteor gen recipe sample -e bigquery -s console
-
-			# generate recipe with multiple sinks
-			$ meteor gen recipe sample -e bigquery -s columbus,kafka -p enrich
-
-			# store recipe to a file
-			$ meteor gen recipe sample -e bigquery -s columbus > recipe.yaml
+			# generate multiple recipes with a template
+			$ meteor gen my-template.yaml -o ./output-dir -d ./data.yaml
 		`),
 		Annotations: map[string]string{
 			"group:core": "true",
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-
-			var sinkList []string
-			var procList []string
-
-			if sinks != "" {
-				sinkList = strings.Split(sinks, ",")
+			templatePath := args[0]
+			bytes, err := ioutil.ReadFile(dataFilePath)
+			if err != nil {
+				return err
 			}
 
-			if processors != "" {
-				procList = strings.Split(processors, ",")
+			var data []recipe.FromTemplateData
+			err = yaml.Unmarshal(bytes, &data)
```

Comment:
Can we handle error like this 
```suggestion
			if err = yaml.Unmarshal(bytes, &data); err != nil {
			     return fmt.Errorf("error unmarshalling bytes: %w", err)
			}
```